source/Plugins.cpp

```diff
@@ -195,11 +196,14 @@ bool Plugin::IsValid() const
 // Attempt to load a plugin at the given path.
 const Plugin *Plugins::Load(const string &path)
 {
+	const bool isZip = path.ends_with(".zip");
 	// Get the name of the folder containing the plugin.
 	size_t pos = path.rfind('/', path.length() - 2) + 1;
-	string name = path.substr(pos, path.length() - 1 - pos);
+	string name = path.substr(pos, path.length() - (isZip ? 4 : 1) - pos);
 
-	string pluginFile = path + "plugin.txt";
+	string realPath = isZip ? path + "/" + Archive::GetRootPath(path) : path;
```

Comment:
This should be abstracted away in Archive; if the archive has a root directory, it should be silently added to the path. Archive handling is not the business of other classes.

Do the GitHub-generated archives come with a root directory? I think whatever they do is the standard we should follow, possibly even require.